id_encoding.md

```diff
@@ -0,0 +1,36 @@
+# Encoding for `feed_id` and `prev_msg_id`
+
+The binary encoding of ids is defined as the concatenation of:
+- the `code` for the type of id, as a UInt8
+- the `value` of the id
+
+## Codes
+
+ code | format       | type |
+:----:|:------------:|:----:|
+ 0    | ed25519      | key  |
+ 1    | sha256       | feed |
+ 2    | sha256.gabby | feed |
```

Comment:
this bit feels like it could be more clear : 
- an id for a feed is currently an ed25519 pk ... but when we say `feedId` are we wanting to capture the type of key, or instead the format of encoding the feed it using 
- an id for a message is ... about how it's signed, how it's packed?

I'm unclear if this table + encoding is fit-for-purpose..

Reply:
Should it instead be like this ? 

 code | type    | feed type | id
:----:|:-------:|:---------:|:---:
 0    | feed_id | classic   | ed25519 pk
 1    | msg_id  | classic   | sha256 hash of message
 0    | feed_id | gabby     | ed25519 pk
 0    | feed_id | gabby     | ???

Reply:
**problem** this magic-number I've made is scuttlebutt-specific ... so the encoding could be ok, but then where's the spec which talk about these codes in scuttlebutt? 

Perhaps I just say "Codes for scuttlebutt"